src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -666,13 +669,26 @@ public boolean databaseExists(final String name) {
     public String error;
   }
 
+  private boolean checkMessagePackSupport() {
+    Matcher matcher = Pattern.compile("(\\d+\\.*)+").matcher(version());
```

Comment:
AFAIK, `Pattern.compile` creates an immutable object so you can reuse it. There is another regular expression that would help you so you don't have to split: `(?:(\d+)\.)(\d+)`. (you can see it working [here](https://www.regexplanet.com/share/index.html?share=yyyyy1bypyr)). You still have to convert it to int, though.